x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+    @Nullable
+    private final String lastSynced;
+    @Nullable
+    private final String name;
+    @Nullable
+    private final ConnectorIngestPipeline pipeline;
+    @Nullable
+    private final ConnectorScheduling scheduling;
+    @Nullable
+    private final String serviceType;
+    @Nullable
+    private final ConnectorStatus status;
+    @Nullable
+    private final Object syncCursor;
+    @Nullable
+    private final boolean syncNow;
+
+    private Connector(
+        String connectorId,
+        String apiKeyId,
+        Map<String, Object> configuration,
+        ConnectorCustomSchedule customScheduling,
+        String description,
+        String error,
+        ConnectorFeatures features,
+        String indexName,
+        boolean isNative,
+        String language,
+        String lastAccessControlSyncError,
+        String lastAccessControlSyncScheduledAt,
+        ConnectorSyncStatus lastAccessControlSyncStatus,
+        Long lastDeletedDocumentCount,
+        String lastIncrementalSyncScheduledAt,
+        Long lastIndexedDocumentCount,
+        String lastSeen,
+        String lastSyncError,
+        String lastSyncScheduledAt,
+        ConnectorSyncStatus lastSyncStatus,
+        String lastSynced,
+        String name,
+        ConnectorIngestPipeline pipeline,
+        ConnectorScheduling scheduling,
+        String serviceType,
+        ConnectorStatus status,
+        Object syncCursor,
+        boolean syncNow
+    ) {
+        this.connectorId = connectorId;
+        this.apiKeyId = apiKeyId;
+        this.configuration = configuration;
+        this.customScheduling = customScheduling;
+        this.description = description;
+        this.error = error;
+        this.features = features;
+        this.indexName = indexName;
+        this.isNative = isNative;
+        this.language = language;
+        this.lastAccessControlSyncError = lastAccessControlSyncError;
+        this.lastAccessControlSyncScheduledAt = lastAccessControlSyncScheduledAt;
+        this.lastAccessControlSyncStatus = lastAccessControlSyncStatus;
+        this.lastDeletedDocumentCount = lastDeletedDocumentCount;
+        this.lastIncrementalSyncScheduledAt = lastIncrementalSyncScheduledAt;
+        this.lastIndexedDocumentCount = lastIndexedDocumentCount;
+        this.lastSeen = lastSeen;
+        this.lastSyncError = lastSyncError;
+        this.lastSyncScheduledAt = lastSyncScheduledAt;
+        this.lastSyncStatus = lastSyncStatus;
+        this.lastSynced = lastSynced;
+        this.name = name;
+        this.pipeline = pipeline;
+        this.scheduling = scheduling;
+        this.serviceType = serviceType;
+        this.status = status;
+        this.syncCursor = syncCursor;
+        this.syncNow = syncNow;
+    }
+
+    public Connector(StreamInput in) throws IOException {
+        this.connectorId = in.readString();
+        this.apiKeyId = in.readOptionalString();
+        if (in.readBoolean()) {
```

Comment:
Why not call [readOptionalMap](https://github.com/elastic/elasticsearch/blob/9063088350388e5318be852730eb1560c1682f7a/server/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java#L397) here?